lib/Doctrine/Node/NestedSet/PreOrderIterator.php

```diff
@@ -105,16 +105,20 @@ public function __construct($record, $opts)
             $query = $q->where("$componentName.lft > ? AND $componentName.rgt < ?", $params)->orderBy("$componentName.lft asc");
         }
 
-        $query = $record->getTable()->getTree()->returnQueryWithRootId($query, $record->getNode()->getRootValue());
+        /** @var Doctrine_Tree_NestedSet $tree */
+        $tree = $record->getTable()->getTree();
+        /** @var Doctrine_Node_NestedSet $node */
+        $node = $record->getNode();
+        $query = $tree->returnQueryWithRootId($query, $node->getRootValue());
 
-        $this->maxLevel   = isset($opts['depth']) ? ($opts['depth'] + $record->getNode()->getLevel()) : 0;
+        $this->maxLevel   = isset($opts['depth']) ? ($opts['depth'] + $node->getLevel()) : 0;
         $this->options    = $opts;
         $this->collection = isset($opts['collection']) ? $opts['collection'] : $query->execute();
         $this->keys       = $this->collection->getKeys();
         $this->count      = $this->collection->count();
         $this->index      = -1;
-        $this->level      = $record->getNode()->getLevel();
-        $this->prevLeft   = $record->getNode()->getLeftValue();
+        $this->level      = $node->getLevel();
+        $this->prevLeft   = $node->getLeftValue();
```

Comment:
This was all changed just to be able to hint to phpstan/psalm what the return of `$record->getNode()` will be (maybe saves a little processing time since `getNode` won't be called multiple times now).